update E5 dataset access

fnattino opened the pull request:
Closes #2

fnattino commented:
Hey @SarahAlidoost, since you already had a look at this episode, would you mind checking how do you see it now as first after the background episodes (new episode 5)?

SarahAlidoost commented on _episodes/19-access-data.md:
```diff
@@ -1,7 +1,7 @@
+teaching: 60
+exercises: 15
 questions:
 - "Where can I find open satellite data?"
```
```suggestion
- "Where can I find open-access satellite data?"
```

SarahAlidoost commented on _episodes/19-access-data.md:
```diff
@@ -10,12 +10,12 @@ objectives:
 - "Search public STAC repositories of satellite imagery using Python."
 - "Inspect the search result metadata."
```
```suggestion
- "Inspect search result's metadata."
```

SarahAlidoost commented on _episodes/19-access-data.md:
```diff
@@ -10,12 +10,12 @@ objectives:
 - "Inspect the search result metadata."
 - "Download (a subset of) the assets available for a satellite scene."
-- "Open the satellite imagery as raster data and setup some raster calculations."
+- "Open the satellite imagery as raster data."
```
```suggestion
- "Open a satellite imagery as raster data."
```

SarahAlidoost commented on _episodes/19-access-data.md:
```diff
@@ -10,12 +10,12 @@ objectives:
 - "Download (a subset of) the assets available for a satellite scene."
-- "Open the satellite imagery as raster data and setup some raster calculations."
+- "Open the satellite imagery as raster data."
 - "Save the processed satellite data to disk."
```
```suggestion
- "Save satellite data to disk."
```

fnattino replied:
This point actually became quite short, so I'll merge it with the previous one!

SarahAlidoost commented on _episodes/19-access-data.md:
```diff
@@ -10,12 +10,12 @@ objectives:
+- "`rioxarray` allows you to open and download remote raster files."
 ---
 
 A number of satellites take snapshots of the Earth's surface from space. The images recorded by these remote sensors
```
```suggestion
# Introduction

A number of satellites take snapshots of the Earth's surface from space. The images recorded by these remote sensors
```

SarahAlidoost reviewed (changes requested):
@fnattino nice addition, well done :+1: I left some minor comments on the changed files, I also committed some formatting and editorial fixes, as we discussed.

I noticed that the lesson does not include a solution for the first exercise. Can we provide solutions, for example some screenshots of the results?
Also, can we replace the link to the github spec specification with the link to its readme i.e. https://github.com/radiantearth/stac-spec#readme ?

fnattino commented:
Thanks @SarahAlidoost - and good point about the solution to the first episode. I'll add some details there!

fnattino commented:
Solution with screenshots added - how do you see it @SarahAlidoost ?

SarahAlidoost reviewed (approved):
@fnattino thanks for addressing my comments. very nice job :+1:  As we discussed, we can still add something like "this is an example and based on what item you selected, the image in the preview can be different" to the solution.  I approved this PR.

cpranav93 commented on _episodes/19-access-data.md:
```diff
@@ -172,50 +196,48 @@ print(item.properties)
-> > from shapely.geometry import Point
-> > p = Point(lon, lat)
-> > bbox = p.buffer(0.05).bounds
+> > bbox = point.buffer(0.01).bounds
```
Certain command such as bbox and query used in this exercise may not be known to all the participants. Could we maybe add hints to these commands or links?

fnattino replied:
Good point, I will add a hint for the `query` argument. About the bbox with the buffer, this is just one way to do it - it is not a requirement and it could be fun to show it while showing the solution.

fnattino commented:
Hi @rbavery , this episode is also ready if/when you have time to review it! 

Since this episode will become the first of the lesson, I have simplified it a bit, removing things that will be introduced later like cropping a raster. 

The most important change is in the last exercise, which used to make use of Landsat data from the Earth Search STAC endpoint. I have noticed that the datasets are no longer available, so I switched STAC endpoint to the NASA CMR one and modified the exercise to ask for Landsat metadata from the HLS collection. I think introducing this second STAC endpoint might actually be a good thing, since this will be used also in the stackstac/odc-stac episode and it will illustrate how public catalogs can also include protected data.

rbavery commented on _episodes/19-access-data.md:
```diff
@@ -384,44 +389,41 @@ visual_clip.rio.to_raster("amsterdam_tci.tif", driver="COG")
 > {: .solution}
 {: .challenge}
 
+> ## Public catalogs, protected data
```
Thanks for these good setup notes. I'm thinking we should also include these ahead of any future episode using the CMR stac's assets that require the instructions, and that the instructor during a workshop may walk the students through this setup. Do you have details on why the os env variable is required here? I don't think I've needed to set that before but I'll test it out later.

fnattino replied:
Hi @rbavery, yep, we could then link back to these instructions from future episodes using NASA CMR STAC or copying instructions there. 

Concerning the GDAL options - I am not sure about the detail, as far as I understand these are used to enable the usage of cookie files that gdal/curl need for authentication. On my system, I was not able to fetch data without setting those (either via env variables or via rasterio.Env with something like `with rasterio.Env(GDAL_HTTP_COOKIEFILE="./cookies.txt", ...)`), I am curious to know whether it works on your side. I've noticed also  @srmsoumya sets them in [here](https://github.com/carpentries-incubator/geospatial-python/pull/102).

rbavery replied:
got it, I also had to do all three to be able to access the actual data and not just the metadata.

rbavery reviewed (changes requested):
Hey @fnattino this looks awesome. Can you update the url to the episode and the table of contents so that this is now lesson 5? I'll give this a more thorough run through tomorrow.

rbavery reviewed (approved):
Great stuff @fnattino . The only thing I'm noticing is that the notebook errors and is using a different collection from the markdown lesson: `collection = "landsat-8-l1-c1"`. can this nb be updated to reflect the markdown? or feel free to delete the nb. I don't mind having the notebooks tracked in vc for convenience, but also it may be easier to maintain just one source of truth for the lessons.

fnattino commented:
Great @rbavery, and thanks for pointing out the issue of the notebook. I will remove it for now, if we see the need for it we could then add at once notebooks for all episodes at once.
